File: Medium/2094-remove-stones-to-minimize-the-total/2094-remove-stones-to-minimize-the-total.py
class Solution:
    def minStoneSum(self, piles: List[int], k: int) -> int:
        #init variable for heap. Build min heap
        max_heap = []
        for pile in piles:
            heapq.heappush(max_heap, -pile)

        #perform operations k number of times
        for i in range(k):
            stone = -(heapq.heappop(max_heap))
            stone = stone - math.floor(stone / 2)
            heapq.heappush(max_heap, -stone)

        return -(sum([num for num in max_heap]))

File: Medium/2094-remove-stones-to-minimize-the-total/2094-remove-stones-to-minimize-the-total.cpp
class Solution {
public:
    int minStoneSum(vector<int>& piles, int k) {
        priority_queue<int>nans(piles.begin(),piles.end());
        for(int i=0;i<k;i++){
            int a = nans.top();
            nans.pop();
            a = a - (a/2);
            nans.push(a);
        }
        int result = 0;
        for(int i=0;i<piles.size();i++){
            int a= nans.top();
            result = result + a;
            nans.pop();
        }
        return result;
    }
};